### PR TITLE
feat: add support for laravel 7 and 8

### PR DESCRIPTION
**Pull request by lakuapik**
This PR not just adding support for laravel 7 and 8, but also refactoring the core and improve documentation.

@axmad386  please kindly review and discuss features here.

### CODE REVIEW

File: src/Exceptions/ApiException.php [hunk 1]
```diff
@@ -2,18 +2,60 @@
 
 namespace KodePandai\ApiResponse\Exceptions;
 
-use Exception;
+use Illuminate\Contracts\Support\Renderable;
 use Illuminate\Contracts\Support\Responsable;
+use Illuminate\Http\JsonResponse;
```
Comment (axmad386) [inline src/Exceptions/ApiException.php]: can we delete this?

File: tests/ApiExceptionTest.php [hunk 2]
```diff
@@ -3,97 +3,100 @@
 use Illuminate\Http\Request;
 use Illuminate\Http\Response;
 use Illuminate\Support\Facades\Route;
+use Illuminate\Support\Facades\Validator;
```
Comment (axmad386) [inline tests/ApiExceptionTest.php]: this unused import removed automatically by lint, but file `AppException.php` is not automatically linted. Should we check php-cs-fixer setup?

**Review by axmad386 — COMMENTED** [review]
Awesome work @lakuapik. I want to discuss something
-  I think `php-cs-fixer` didn't work properly. Check my comments.
- how about laravel 5 and 6? it is possible to support them?
- Should we include documentation how to customize response structure? I think we need to mention `toResponse` method part

**Comment by lakuapik** [comment]
@axmad386 i run `compsoser test` again.

```txt
❯ date; git branch --show-current; composer test
Mon Jun 13 10:58:00 AM WIB 2022
support-laravel-7-and-8
> php-cs-fixer fix --allow-risky=yes
Loaded config default from "/media/d/data/work/labs/laravel-api-response/.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".

Fixed all files in 0.006 seconds, 16.000 MB memory used
> pest --stop-on-failure

   PASS  Tests\ApiExceptionTest
  ✓ it can handle ApiException
  ✓ it returns HTTP_BAD_REQUEST for ApiException by default
  ✓ it can handle ApiValidationException
  ✓ it can access route with validated parameters

   PASS  Tests\ApiResponseTest
  ✓ it returns correct response header
  ✓ it returns correct json structure for success api response
  ✓ it returns correct json structure for error api response
  ✓ it fails when the argument $data is not valid
  ✓ it fails when the argument $errors is not valid

  Tests:  9 passed
  Time:   0.41s
```

Off course we could also support laravel 5 and 6, but we need to ditch pestphp and move to phpunit. pest require minimum version of php 7.3. which is doable for laravel 7. for laravel 5 and 6, it can run below 7.3.

**Comment by axmad386** [comment]
> Off course we could also support laravel 5 and 6, but we need to ditch pestphp and move to phpunit. pest require minimum version of php 7.3. which is doable for laravel 7. for laravel 5 and 6, it can run below 7.3.

Oh I see. Then it's better to exclude laravel 5 and 6 for now.

**Review by axmad386 — APPROVED** [review]
Everything is ok, let's merge this to the main branch

**Pull request merged**